sources/PTYTextView.m

```diff
@@ -1560,40 +1560,52 @@ - (BOOL)scrollWheelShouldSendArrowForEvent:(NSEvent *)event at:(NSPoint)point {
         return NO;
     }
     BOOL alternateMouseScroll = [iTermAdvancedSettingsModel alternateMouseScroll];
+    NSString* alternateMouseScrollEscapeUp = [iTermAdvancedSettingsModel alternateMouseScrollEscapeUp];
+    NSString* alternateMouseScrollEscapeDown = [iTermAdvancedSettingsModel alternateMouseScrollEscapeDown];
     BOOL showingAlternateScreen = [self.dataSource showingAlternateScreen];
-    if (!alternateMouseScroll && showingAlternateScreen) {
-        [_altScreenMouseScrollInferer scrollWheel:event];
-    }
-    if (!alternateMouseScroll) {
+    if (!showingAlternateScreen) {
         return NO;
     }
-    if (!showingAlternateScreen) {
+    
+    if (alternateMouseScroll) {
+        *upData = [_dataSource.terminal.output keyArrowUp:event.modifierFlags];
+        *downData = [_dataSource.terminal.output keyArrowDown:event.modifierFlags];
+        return YES;
+    } else if (alternateMouseScrollEscapeUp.length && alternateMouseScrollEscapeDown.length) {
+        
+        NSString* e_ups = [NSString stringWithFormat:@"\e%@", alternateMouseScrollEscapeUp ];
```

Comment:
This should probably send multiple repeats of the code when you scroll fast. Also, don't prepend `\e`. Instead, use `-[NSString stringByExpandingVimSpecialCharacters]`, which converts a `\e` into a <kbd>esc</kbd>.